### PR TITLE
Archive containerd-supported images tarballs

### LINKED ISSUE #1250: Tar container images format not supported by containerd
**Component**: build

**What happened**:
The format of the tar container images is not supported by containerd thus import of pause and nginx fails silently
**What was expected**:
The images are imported and injected into the containerd cache
**Steps to reproduce**
run the `ctr --debug images import xxx.tar`, adding the debug flag will show the error
**Resolution proposal** (optional):
Archive the images with docker save for the time being as the format is accepted by containerd

### PR DESCRIPTION
**Pull request by sayf-eddine-scality**
Fixes: #1250

**Component**:
build

**Context**: 
container image build
**Summary**:
Skopeo produces incompatible tarballs with containerd thus we re
replacing it with docker in that step, the normal image (no tarballs
should stay the same)
**Acceptance criteria**: 
containerd should be able to import the produced images (nginx and pause)

---

Closes: #1250 

See: #942

### CODE REVIEW

**Review by TeddyAndrieux — DISMISSED** [review]

File: buildchain/buildchain/targets/remote_image.py [hunk 1]
```diff
@@ -101,6 +112,15 @@ def task(self) -> types.TaskDict:
             })
         return task
 
+    def _docker_pull(self) -> List[str]:
```
Comment (slaperche-scality) [inline buildchain/buildchain/targets/remote_image.py]: No, we don't want to use `docker` directly anymore.

We should use the API, @jbertran did the work.

You can restore the code that was removed from my PR.

**Review by slaperche-scality — CHANGES_REQUESTED** [review]

File: buildchain/buildchain/targets/remote_image.py (hunk 1 above)
Comment (slaperche-scality) [reply buildchain/buildchain/targets/remote_image.py]: See 396d60824825b9fb7792d1e18e8be3504fc2c24e

File: buildchain/buildchain/targets/remote_image.py [hunk 2]
```diff
@@ -92,7 +103,7 @@ def task(self) -> types.TaskDict:
         })
         if self._use_tar:
             task.update({
-                'actions': [self._skopeo_copy()],
+                'actions': [self._docker_pull(), self._docker_save()],
```
Comment (slaperche-scality) [inline buildchain/buildchain/targets/remote_image.py]: IIRC, we need a `tag` between the `pull` and the `save`.

**Comment by NicolasT** [comment]
So, I assume `skopeo`s `oci-archive` output doesn't cut it? :disappointed: (CC @rhatdan)

**Comment by sayf-eddine-scality** [comment]
Yes the image tarball is made and ctr doesnt complain about any error but it says `unpacking 0 images` ...

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Comment by sayf-eddine-scality** [comment]
@bert-e approve

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

File: buildchain/buildchain/docker_command.py [hunk 3]
```diff
@@ -237,3 +237,66 @@ def __call__(self) -> Optional[TaskError]:
             command=self.command,
             **run_config
         )
+# The call method is not counted as a public method
```
Comment (slaperche-scality) [inline buildchain/buildchain/docker_command.py]: Skip two lines between `DockerRun` and `DockerTag`.

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

File: buildchain/buildchain/docker_command.py [hunk 4]
```diff
@@ -237,3 +237,66 @@ def __call__(self) -> Optional[TaskError]:
...
+# pylint: disable=too-few-public-methods
```
Comment (slaperche-scality) [inline buildchain/buildchain/docker_command.py]: To disable the warning only for the class, the correct syntax is:

```
class DockerTag:
    # pylint: disable=too-few-public-methods
    """A class to expose the `docker tag` command through the API client."""
```

(yeah it was wrong before, we noticed later in another Jordi's PR).

**Review by slaperche-scality — DISMISSED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Review by slaperche-scality — APPROVED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e approve

**Pull request merged**